pkg/vadmin/interface.go

```diff
@@ -138,6 +143,34 @@ func MakeVClusterOps(log logr.Logger, vdb *vapi.VerticaDB, cli client.Client, vo
 	}
 }
 
+// prepLocalData Prepare for the add node or create_db by removing any local
+// data/depot dirs and ensuring proper ownership.
+// This step is necessary because of a lack of cleanup in admintools if any of
+// these commands fail.
+func PrepLocalData(ctx context.Context, vdb *vapi.VerticaDB, prunner cmds.PodRunner, podName types.NamespacedName) error {
```

Comment:
Lets not add a new function for this to the interface. We aren't treating it like a true abstraction, since we only call it for the admintools based deployments. Rather, lets just add the prep calls to the admintools versions of CreateDB, ReviveDB and AddNode. We may need to include additional options for these API calls. We know the hosts that we are using but not necessarily the pod names.

Reply:
Thank you for your suggestion. As you said, We may need to include additional options for these API calls. PrepLocalData needs podName, but we didn't have podName in create_db and revive_db. How do we know the pod names from the hosts?

Reply:
You have to add it to the options then have the operator fill that in ahead of calling the dispatcher interface.

Reply:
I got a strange error message after I try your suggestion:  pods `v-enterprise-sanity-sc1-0` is forbidden.' @spilchen , do you have any insights into this?
```
kuttl-test-complete-koi verticadb-operator-controller-manager-ff7f545d4-k5xj6 manager 2023-09-21T01:38:54.884Z	ERROR	Reconciler error	{"controller": "verticadb", "controllerGroup": "vertica.com", "controllerKind": "VerticaDB", "VerticaDB": {"name":"v-enterprise-sanity","namespace":"kuttl-test-complete-koi"}, "namespace": "kuttl-test-complete-koi", "name": "v-enterprise-sanity", "reconcileID": "d9eca9cb-4e6c-4543-928d-29a0b64d1be8", "error": "could not execute: pods \"v-enterprise-sanity-sc1-0\" is forbidden: User \"system:serviceaccount:kuttl-test-complete-koi:verticadb-operator-controller-manager\" cannot create resource \"pods/exec\" in API group \"\" at the cluster scope"}
```